### DISCUSSION TITLE
Strategies for Distributing R Packages to Clients: Balancing Open Source and Intellectual Property

### DISCUSSION THREAD

**Discussion by nturaga** [rinpharma/rinpharma-summit-2024#22] [roundtable]
### Proposal

This roundtable discussion will explore the challenges and solutions for pharmaceutical organizations developing and distributing R packages to their clients. We'll focus on:

1. Methods for sharing R packages while protecting proprietary code and algorithms
2. Legal considerations and licensing options for R packages in Pharma / Biotech
3. Technical approaches to limit code visibility (e.g., compiled code, remote APIs)
4. Best practices for documentation and support when distributing limited-access packages
5. Balancing open-source contributions with IP protection in the R ecosystem

### Expected impact

This discussion could benefit our community by:

1. Providing practical strategies for companies to share valuable R tools while safeguarding their intellectual property
2. Fostering collaboration between legal, IT, and R developers to create robust distribution policies
3. Exploring innovative technical solutions that serve both open-source principles and business needs
4. Addressing a common challenge faced by many pharmaceutical companies using R
5. Potentially leading to the development of new tools or best practices for secure R package distribution

### Prior discussions/work

While this topic hasn't been extensively addressed in the context of pharmaceutical R packages, there are related discussions and resources we can build upon:

1. The development of package management systems like 'packrat' and 'renv' for creating reproducible and isolated R environments, which could be adapted for secure distribution.
2. Discussions on creating APIs from R functions, such as the 'plumber' package, which allows R functions to be exposed as web APIs. This approach could be used to provide access to functionality without exposing the underlying code.
3. The ongoing debate in the R community about compiled vs. interpreted code, and its implications for performance and code protection. This includes discussions about the “Rcpp” package and its use in creating compiled components within R packages.
4. Industry case studies on using Docker containers to distribute R-based analytics tools, providing a layer of isolation and potential security.
5. Discussions in the broader software community about code obfuscation techniques and their applicability to R. Obfuscation might complement but doesn't replace proper licensing and legal protections.

This roundtable aims to bring these various threads together, focusing specifically on the needs and constraints of the pharmaceutical industry. We'll explore how these approaches might be combined or adapted to create secure, distributable R packages that protect intellectual property while still providing value to clients.

### Would you be willing to potentially facilitate this discussion?

Maybe/No/ask me again later

**Comment by epijim** [comment]
adde to agenda as-is
